scripts/events_module/event_filters.py

```diff
@@ -0,0 +1,412 @@
+import re
+
+import ujson
+
+from scripts.game_structure.game_essentials import game
+from scripts.utility import (
+    get_alive_status_cats,
+    filter_relationship_type,
+)
+
+
+def event_for_location(locations: list) -> bool:
+    """
+    checks if the clan is within the given locations
+    """
+    if "any" in locations:
+        return True
+
+    for place in locations:
+        if ":" in place:
+            info = place.split(":")
+            req_biome = info[0]
+            req_camps = info[1].split("_")
+        else:
+            req_biome = place
+            req_camps = ["any"]
+
+        if req_biome == game.clan.biome.lower():
+            if "any" in req_camps or game.clan.camp_bg in req_camps:
+                return True
+            else:
+                return False
+            
+        return False
+
+
+def event_for_season(seasons: list) -> bool:
+    """
+        checks if the clan is within the given seasons
+        """
+    if "any" in seasons:
+        return True
+    elif game.clan.current_season.lower() in seasons:
+        return True
+
+    return False
+
+
+def event_for_tags(tags: list, cat, other_cat=None) -> bool:
+    """
+        checks if current tags disqualify the event
+        """
+    if not tags:
+        return True
+
+    # some events are mode specific
+    mode = game.clan.game_mode
+    possible_modes = ["classic", "expanded", "cruel_season"]
+    for _poss in possible_modes:
+        if _poss in tags and mode != _poss:
+            return False
+
+    # check leader life tags
+    if cat.status == "leader":
+        leader_lives = game.clan.leader_lives
+
+        life_lookup = {
+            "some_lives": 4,
+            "lives_remain": 2,
+            "high_lives": 7,
+            "mid_lives": 4,
+            "low_lives": 1
+        }
+
+        for _con, _val in life_lookup.items():
+            if _con in tags and leader_lives < _val:
+                return False
+
+    # check for required ranks within the clan
+    for _tag in tags:
+        rank_match = re.match(r"clan:(.+)", _tag)
+        if not rank_match:
+            continue
+        ranks = [x for x in rank_match.group(1).split(",")]
+
+        for rank in ranks:
+            if rank == "apps":
+                if not get_alive_status_cats(
+                        cat,
+                        ["apprentice", "medicine cat apprentice", "mediator apprentice"]):
+                    return False
+                else:
+                    continue
+
+            if rank in ["leader", "deputy"] and not get_alive_status_cats(cat, [rank]):
+                return False
+
+            elif not len(get_alive_status_cats(cat, [rank])) >= 2:
```

Comment:
```suggestion
            if not len(get_alive_status_cats(cat, [rank])) >= 2:
```
No way to get here without `continue` or `return`, so it's okay to just be an `if`